sycl/include/CL/sycl/detail/handler_proxy.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <CL/sycl/access/access.hpp>
+#include <CL/sycl/detail/export.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+
+class handler;
+
+namespace detail {
+
+class AccessorBaseHost;
+
+__SYCL_EXPORT void associateWithHandler(handler &, AccessorBaseHost *,
```

Comment:
Just a question: does this function need __SYCL_EXPORT? wouldn't it be better to make it DLL_LOCAL?

Reply:
This function is referred to from `accessor`'s constructor, which is template. Hence, the user's binary will have a reference to it.

Reply:
Interesting.
@alexbatashev please take a look at this case.
IMO, it seems pretty similar to what you did in your PR here: https://github.com/intel/llvm/pull/1768
For example, handleHostData() was marked with DLL_LOCAL.
handleHostData() (sycl_mem_obj_t.hpp ) is called from buffer_impl constructor (buffer_impl.hpp), called from buffer constructor.

Reply:
@v-klochkov that's different. In my case all functions are in header files. So, even if they have private linkage, they are *always* available to user. There could be another solution: move all that stuff to cpp files to make sure functions are always exported. But that would prevent compiler from optimizing most of the calls (that is especially sad for one-liners).

In Sergey's case, the code is in the *.cpp file, so he has to export the function. I believe, there's no way to break the dependency without moving some stuff to the library. Am I right, @s-kanaev?

Reply:
> I believe, there's no way to break the dependency without moving some stuff to the library.

That's a precise description of my approach.

Reply:
Right. I forgot that the function was in handler_proxy.cpp (I wrongly thought it was in handler_proxy.hpp).
Thank you.